Browser/test.fsx

```diff
@@ -0,0 +1,87 @@
+#load "Fable.Import.WebWorker.fs"
+
+open Fable.Import.WebWorker
+open Fable.Import.WebWorker.ServiceWorker
+
+self.addEventListener_install(fun installEvent ->
+    printfn "service worker installed, %A" installEvent
+    installEvent.waitUntil(self.skipWaiting())
+)
+
+self.addEventListener_activate(fun activateEvent ->
+    printfn "service worker activated, %A" activateEvent
+    activateEvent.waitUntil(self.clients.claim())
+)
+
+type MyPushNotificationInfo = unit
+
+self.addEventListener_push (fun ev ->
+    let json = 
+        ev.data
+        |> Option.map (fun data -> data.json())
+
+    match json with
+    | Some payload ->
+        let myInfo = payload :?> MyPushNotificationInfo
+        
+        let title = "My title"
+        let options:NotificationOptions =
```

Comment:
How should this be tackled? What should NotificationOptions be in this case? Just a plain record?

Reply:
When facing this kind of situation I like to provide an helper on top of it.

Something like:

```
Notification.empty
|> Notification.body "my body"
|> Notification.icon "my-icon"
....
```

Like that the user only specify what is needed for him. This is the same approach I used in [`Thoth.Elmish.Toast`](https://mangelmaxime.github.io/Thoth/elmish/toast_demo.html) for example.